arcgis-ios-sdk-samples/Maps/Apply scheduled updates to preplanned map area/ApplyScheduledUpdatesToPreplannedMapAreaViewController.swift

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright © 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+/// A view controller that manages the interface of the Apply Scheduled Updates
+/// to Preplanned Map Area sample.
+class ApplyScheduledUpdatesToPreplannedMapAreaViewController: UIViewController {
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView!
+    
+    /// The mobile map package used by this sample.
+    let mobileMapPackage: AGSMobileMapPackage!
+    /// The sync task used to check for scheduled updates.
+    var offlineMapSyncTask: AGSOfflineMapSyncTask!
+    /// The sync job used to apply updates to the offline map.
+    var offlineMapSyncJob: AGSOfflineMapSyncJob!
+    
+    required init?(coder: NSCoder) {
+        let mobileMapPackageURL = Bundle.main.url(forResource: "canyonlands", withExtension: nil)!
+        do {
+            let temporaryDirectoryURL = try FileManager.default.url(
+                for: .itemReplacementDirectory,
+                in: .userDomainMask,
+                appropriateFor: mobileMapPackageURL,
+                create: true
+            )
+            let temporaryMobileMapPackageURL = temporaryDirectoryURL.appendingPathComponent(ProcessInfo.processInfo.globallyUniqueString)
+            try FileManager.default.copyItem(at: mobileMapPackageURL, to: temporaryMobileMapPackageURL)
+            mobileMapPackage = AGSMobileMapPackage(fileURL: temporaryMobileMapPackageURL)
+        } catch {
+            print("Error setting up mobile map package: \(error)")
+            mobileMapPackage = nil
+        }
+        
+        super.init(coder: coder)
+        
+        mobileMapPackage?.load { [weak self] (error) in
```

Comment:
`mobileMapPackage` is defined as non-optional.  Does the compiler give a warning if you leave off the `?`?

Reply:
It is defined as an implicitly unwrapped optional (IUO). The only way it will be `nil` is if creation of the temporary copy fails, here in the initializer. So we need to use optional chaining on the off chance that happened. We don't need to use optional chaining elsewhere as the only way that code can be reached is if the mobile map package was loaded and therefore exists. Does that make sense?